From: Will Thompson <wjt@endlessm.com>
Date: Wed, 22 Aug 2018 17:32:55 +0100
Subject: docs: make FlatpakTransactionOperation appear

There were two reasons why no docs for this class appeared in the HTML
documentation:

* flatpak-transaction-operation.xml was not included by
  flatpak-docs.xml.
* all its symbols were listed in <SUBSECTION Standard>, which is hidden
  from the HTML output. It appears that if a section has no visible
  symbols, it's omitted.

Many symbols which belong to FlatpakTransaction were erroneously grouped
with FlatpakTransactionOperation and so also hidden; fix this too.

Closes: #1995
Approved by: alexlarsson

(cherry picked from commit 98ee688b849146723d1808ee662a091500e85743)
---
 doc/reference/flatpak-docs.xml     |  1 +
 doc/reference/flatpak-sections.txt | 25 +++++++++++++++----------
 2 files changed, 16 insertions(+), 10 deletions(-)

diff --git a/doc/reference/flatpak-docs.xml b/doc/reference/flatpak-docs.xml
index 8a3ec2b..555fad2 100644
--- a/doc/reference/flatpak-docs.xml
+++ b/doc/reference/flatpak-docs.xml
@@ -17,6 +17,7 @@
     <title>Flatpak</title>
     <xi:include href="xml/flatpak-installation.xml"/>
     <xi:include href="xml/flatpak-remote.xml"/>
+    <xi:include href="xml/flatpak-transaction-operation.xml"/>
     <xi:include href="xml/flatpak-transaction-progress.xml"/>
     <xi:include href="xml/flatpak-transaction.xml"/>
     <xi:include href="xml/flatpak-ref.xml"/>
diff --git a/doc/reference/flatpak-sections.txt b/doc/reference/flatpak-sections.txt
index 0808740..7d3e94c 100644
--- a/doc/reference/flatpak-sections.txt
+++ b/doc/reference/flatpak-sections.txt
@@ -197,13 +197,7 @@ flatpak_transaction_progress_get_type
 <SECTION>
 <FILE>flatpak-transaction-operation</FILE>
 <TITLE>FlatpakTransactionOperation</TITLE>
-<SUBSECTION Standard>
-flatpak_transaction_add_default_dependency_sources
-flatpak_transaction_add_dependency_source
-flatpak_transaction_add_uninstall
-flatpak_transaction_get_current_operation
-flatpak_transaction_get_operations
-flatpak_transaction_get_installation
+FlatpakTransactionOperation
 flatpak_transaction_operation_get_bundle_path
 flatpak_transaction_operation_get_commit
 flatpak_transaction_operation_get_operation_type
@@ -212,8 +206,12 @@ flatpak_transaction_operation_get_remote
 flatpak_transaction_operation_get_metadata
 flatpak_transaction_operation_get_old_metadata
 flatpak_transaction_operation_type_to_string
+<SUBSECTION Standard>
+FlatpakTransactionOperationClass
+FLATPAK_IS_TRANSACTION_OPERATION
+FLATPAK_TRANSACTION_OPERATION
 FLATPAK_TYPE_TRANSACTION_OPERATION
-FlatpakTransactionOperation
+flatpak_transaction_operation_get_type
 </SECTION>
 
 <SECTION>
@@ -226,10 +224,18 @@ FlatpakTransactionResult
 flatpak_transaction_new_for_installation
 flatpak_transaction_add_install
 flatpak_transaction_add_install_bundle
+flatpak_transaction_add_install_flatpakref
 flatpak_transaction_add_update
-flatpak_transaction_is_empty
+flatpak_transaction_add_uninstall
+flatpak_transaction_add_default_dependency_sources
+flatpak_transaction_add_dependency_source
 flatpak_transaction_run
 <SUBSECTION>
+flatpak_transaction_get_current_operation
+flatpak_transaction_get_installation
+flatpak_transaction_get_operations
+flatpak_transaction_is_empty
+<SUBSECTION>
 flatpak_transaction_set_disable_dependencies
 flatpak_transaction_set_disable_prune
 flatpak_transaction_set_disable_related
@@ -239,7 +245,6 @@ flatpak_transaction_set_no_pull
 flatpak_transaction_set_reinstall
 flatpak_transaction_set_force_uninstall
 flatpak_transaction_set_default_arch
-flatpak_transaction_add_install_flatpakref
 <SUBSECTION Standard>
 FlatpakTransactionClass
 FLATPAK_IS_TRANSACTION
